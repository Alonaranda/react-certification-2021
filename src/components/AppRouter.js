import React from 'react';
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { VideoDetailsScreen } from '../components/VideoDetailsScreen';
import { HomeView } from './HomeView';
import { LoginView } from './LoginView';
import { NotFound } from './NotFound';

export const AppRouter = () => {

    return (
        <Router>
            <>
                <Switch>
                    <Route exact path="/login" component={LoginView} />

                    <Route exact path="/" component={HomeView} />
                    <Route exact path="/videodetails" component={VideoDetailsScreen} />

                    <Route path="*" component={NotFound} />

                </Switch>
            </>
        </Router>
    )
}
