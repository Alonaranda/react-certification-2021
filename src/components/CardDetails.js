import React from 'react'

export const CardDetails = ({
    id,
    videoId,
    title,
    descripcion
}) => {
    return (
        <div>
            <h1>{id}</h1>
            <h2>{videoId}</h2>
            <h3>{title}</h3>
            <h4>{descripcion}</h4>
        </div>
    )
}
