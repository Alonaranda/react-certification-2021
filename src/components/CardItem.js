import React from 'react';
import PropTypes from 'prop-types';
import {Link } from 'react-router-dom';

export const CardItem = ({ title, description, url, videoId, categoria }) => {


  return (

    <div className="card-item">
      <img src={url} alt={title}></img>
      <h2>{title}</h2>
      <p>{description}</p>
      <button
      ><Link
        to={`/videoDetails/${videoId}&${categoria}`}
      ><i className="far fa-play-circle"></i></Link></button>

    </div>

  );

};

CardItem.prototype = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  videoId: PropTypes.string.isRequired
};
