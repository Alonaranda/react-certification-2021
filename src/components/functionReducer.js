

export const functionReducer = (state = [], action) => {
    switch (action.type) {
        case 'addSearch':
            return [...state, action.payload];
    
        default:
            return state;
    }
}