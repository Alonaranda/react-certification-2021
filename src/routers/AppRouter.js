import React, { useContext } from 'react';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { GlobalContext } from '../components/Context';
import { LoginView } from '../components/LoginView';
import { NotFound } from '../components/NotFound';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

    const { myStateReducer } = useContext(GlobalContext)

    return (
        <Router>
            <>
                <Switch>
                    <PublicRoute 
                        exact path="/login" 
                        component={LoginView} 
                        isAuthenticated={myStateReducer.logged}
                        
                    />
                    <PublicRoute>
                        <Route path="*" component={NotFound} />
                    </PublicRoute>

                    <PrivateRoute 
                        path="/" 
                        component={DashboardRoutes} 
                        isAuthenticated={myStateReducer.logged}
                    />

                    

                </Switch>
            </>
        </Router>
    )
}
