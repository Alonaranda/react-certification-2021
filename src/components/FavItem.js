import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import '../style/FavoriteVideosView.css'
import { GlobalContext } from './Context';

export const FavItem = ({ title, description, url, videoId }) => {

    const { myStateReducer, dispatch } = useContext(GlobalContext);

    const handleClickPlay = () => {
        dispatch({
            type: 'actionAddVideoDataFav',
            payload: {
                ...myStateReducer,
                videoDataFav: {
                    title: title,
                    description: description,
                    url: url,
                    videoId: videoId
                }
            }
        });
    }

    const handleClickDelete = () => {
        dispatch({
            type: 'actionDelete',
            payload: videoId
        });
    }


    return (
        <div className="card-item">
            <img src={url} alt={title}></img>
            <h2>{title}</h2>
            <p>{description}</p>
            <div className="botones-favoritos">
                <button
                    className="button-left"
                    onClick={handleClickPlay}
                ><Link
                    to={`/videodetailsfavs`}
                ><i className="far fa-play-circle"></i></Link></button>

                <button
                    className="button-right"
                    onClick={handleClickDelete}
                ><i className="far fa-trash-alt"></i></button>


            </div>
        </div>
    );
};

FavItem.prototype = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    videoId: PropTypes.string.isRequired,
};
