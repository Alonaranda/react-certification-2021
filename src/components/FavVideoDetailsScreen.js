import React, { useContext } from 'react'
import { GlobalContext } from './Context'
import { FavItem } from './FavItem'
import { HeaderNav } from './HeaderNav'

export const FavVideoDetailsScreen = () => {

    const { myStateReducer } = useContext(GlobalContext)

    return (
        <>
            <div className="favorite-header">
                <HeaderNav />
            </div>
            <div className="contenedor">
                <div className="video-details">
                    <div className="video-details-vid">
                        <iframe
                            allowFullScreen
                            frameBorder="0"
                            title="rick roll"
                            src={`https://www.youtube.com/embed/${myStateReducer.videoDataFav.videoId}?autoplay=1`}
                            allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
                        />

                    </div>

                    <div className="video-details-text">
                        <h2>{myStateReducer.videoDataFav.title}</h2>
                        <p>{myStateReducer.videoDataFav.description}</p>
                    </div>
                </div>
                <hr />
            </div>


            <div className="contenedor-sugeridos">
                <h3>Favorites: </h3>
                <div className="sugeridos">
                    {myStateReducer.favorites.map(vid => (
                        <FavItem
                            key={vid.videoId}
                            {...vid}
                        />
                    ))}
                </div>
            </div>


        </>
    )
}
