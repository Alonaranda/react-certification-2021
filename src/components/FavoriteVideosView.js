import React, { useContext } from 'react'
import { GlobalContext } from './Context'
import { FavItem } from './FavItem';
import { HeaderNav } from './HeaderNav';

export const FavoriteVideosView = () => {

    const { myStateReducer } = useContext(GlobalContext);

    return (
        <>
            <div className="favorite-header">
                <HeaderNav />
            </div>
            <div className="contenedor">
                <h1>Favorite Videos:</h1>
                <div className="cards">
                    {
                        myStateReducer.favorites.map(vid => (
                            <FavItem
                                key={vid.videoId}
                                {...vid}
                            />
                        ))
                    }
                </div>
            </div>
        </>
    )
}
