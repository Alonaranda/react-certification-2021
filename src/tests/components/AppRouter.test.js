import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { MemoryRouter, Router } from "react-router";
import { AppRouter } from "../../components/AppRouter";
import { GlobalContext } from "../../components/Context";


describe('Test AppRouter', () => {
    const contextValue = {
        dispatch: jest.fn(),
        myStateReducer: {
            mode: 'light',
            search: 'wizeline',
            videoData: {}
        }
    }
    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const wrapper = mount(
        <GlobalContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <AppRouter/>
                </Router>
            </MemoryRouter>
        </GlobalContext.Provider>
    );

    afterEach(() => {
        jest.clearAllMocks();
    })

    it('should render success', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.exists());
    });
});