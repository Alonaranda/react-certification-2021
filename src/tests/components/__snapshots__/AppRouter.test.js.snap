// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AppRouter should render success 1`] = `ReactWrapper {}`;
